### PR TITLE
Update sidebar & breadcrumbs when editing project

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Updates the project name in the layout sidebar and breadcrumbs when a project name is successfully updated.
Pulled from [#189](https://github.com/phac-nml/irida-next/pull/189).

## Screenshots or screen recordings
Screenshot from 2023-09-21 16-14-36

## How to set up and validate locally
1. Navigate to a Project
2. Click the `Settings` link in the sidebar
3. In the `Project name and description` form set `Project Name` to `a`
4. Click `Update project`
5. See that the form displays an error message and the sidebar and breadcrumbs are not updated.
6. Change the `Project Name` to `New project name`.
7. Click `Update project`
8. See a success message and that the sidebar an breadcrumbs have changed to the new name.

Note: Verify that change path refreshes the page to the new URL as expected.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Comment by deepsidhu85** [comment]
Just leaving a general comment on the PR of the changes we discussed:

1) Use html sytnax to create a turbo frame around the breadcrumbs so we don't need to use a helper in the component

2) After setting the `change_path submit button to not use turbo when the path was updated but there was an error it caused the url to change with the `/-/edit` removed from it. Update this to use turbo and the update method in projects_controller to only respond to turbo_stream

3) Since the breadcrumbs are now in a turbo frame the navigation needs to be updated to it happens "outside" of the turbo frame. This can be updated in the breadcrumb component where the link is created to set the data: {turbo_frame: "_top"} for each of the link_to tags

4) Looks like I accidentally deleted the review I had left, but it was just to updated the name of the form_fields partial to a clearer name

**Review by joshsadam — APPROVED** [review]
Thanks @ksierks this appears to work the way it is supposed to.  Once @ deepsidhu85 gives it the thumbs up, this is good to go.  We still may want to break each section of the settings page out into its own turbo frame at some point, so you don't need to return it all.

**Comment by ksierks** [comment]
> Just leaving a general comment on the PR of the changes we discussed and that were made:
> 
> 1. Use html sytnax to create a turbo frame around the breadcrumbs so we don't need to use a helper in the component
> 2. After setting the `change_path submit button to not use turbo when the path was updated but there was an error it caused the url to change with the `/-/edit` removed from it. Update this to use turbo and the update method in projects_controller to only respond to turbo_stream
> 3. Since the breadcrumbs are now in a turbo frame the navigation needs to be updated to it happens "outside" of the turbo frame. This can be updated in the breadcrumb component where the link is created to set the data: {turbo_frame: "_top"} for each of the link_to tags
> 4. Looks like I accidentally deleted the review I had left, but it was just to updated the name of the form_fields partial to a clearer name

To address #4, I renamed the form `_edit_name_and_description_form.html.erb` like in the groups PR. Please see aaea28541ed40a2e45c9937e8e0a812428e6cb2d.

File: test/system/projects_test.rb [hunk 1]
```diff
@@ -91,8 +91,20 @@ def setup
     fill_in I18n.t(:'activerecord.attributes.namespaces/project_namespace.name'), with: project_name
```
Comment (deepsidhu85) [inline test/system/projects_test.rb:91]: Can I get you to add a test for a successful path change? We will want to test if the url updated correctly and assert that the success text was displayed in the flash

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This all looks great. Just a test to add below

File: test/system/projects_test.rb (hunk 1 above)
Comment (deepsidhu85) [reply test/system/projects_test.rb:91]: You are correct! Totally missed it. Thanks

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**